lib/galaxy/managers/library_datasets.py

```diff
@@ -195,17 +195,26 @@ def serialize(self, trans, ld):
         if len(expired_ldda_versions) > 0:
             rval['has_versions'] = True
             rval['expired_versions'] = expired_ldda_versions
+
+        ldda = ld.library_dataset_dataset_association
+        if ldda.creating_job_associations:
+            if ldda.creating_job_associations[0].job.stdout:
+                rval['job_stdout'] = ldda.creating_job_associations[0].job.stdout.strip()
+            if ldda.creating_job_associations[0].job.stderr:
+                rval['job_stderr'] = ldda.creating_job_associations[0].job.stderr.strip()
+        if ldda.dataset.uuid:
+            rval['uuid'] = str(ldda.dataset.uuid)
         rval['deleted'] = ld.deleted
         rval['folder_id'] = 'F' + rval['folder_id']
         rval['full_path'] = full_path
-        rval['file_size'] = util.nice_size(int(ld.library_dataset_dataset_association.get_size()))
-        rval['date_uploaded'] = ld.library_dataset_dataset_association.create_time.strftime("%Y-%m-%d %I:%M %p")
+        rval['file_size'] = util.nice_size(int(ldda.get_size()))
+        rval['date_uploaded'] = ldda.create_time.strftime("%Y-%m-%d %I:%M %p")
         rval['can_user_modify'] = trans.app.security_agent.can_modify_library_item(current_user_roles, ld) or trans.user_is_admin()
-        rval['is_unrestricted'] = trans.app.security_agent.dataset_is_public(ld.library_dataset_dataset_association.dataset)
-        rval['tags'] = self.tag_manager.get_tags_str(ld.library_dataset_dataset_association.tags)
+        rval['is_unrestricted'] = trans.app.security_agent.dataset_is_public(ldda.dataset)
+        rval['tags'] = self.tag_manager.get_tags_str(ldda.tags)
 
         #  Manage dataset permission is always attached to the dataset itself, not the the ld or ldda to maintain consistency
-        rval['can_user_manage'] = trans.app.security_agent.can_manage_dataset(current_user_roles, ld.library_dataset_dataset_association.dataset) or trans.user_is_admin()
+        rval['can_user_manage'] = trans.app.security_agent.can_manage_dataset(current_user_roles, ldda.dataset) or trans.user_is_admin()
```

Comment:
It may be faster to swap the conditions:
```python
trans.user_is_admin() or trans.app.security_agent.can_manage_dataset(current_user_roles, ldda.dataset)
```

Reply:
Good catch!